Documentation/gettingstarted/cni-chaining-aws-cni.rst

```diff
@@ -0,0 +1,131 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+*******
+AWS-CNI
+*******
+
+This guide explains how to set up Cilium in combination with aws-cni. In this
+hybrid mode, the aws-cni plugin is responsible for setting up the virtual
+network devices as well as address allocation (IPAM) via ENI. After the initial
+networking is setup, the Cilium CNI plugin is called attach BPF programs to the
+network devices set up by aws-cni to enforce network policies, perform
+load-balancing, and encryption.
+
+.. image:: aws-cni-architecture.png
+
+
+Setup Cluster on AWS
+====================
+
+Follow the instructions in the :ref:`k8s_install_eks` guide to set up an EKS
+cluster or use any other method of your preference to set up a Kubernetes
+cluster.
+
+Ensure that the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_
+plugin is installed. If you have set up an EKS cluster, this is automatically
+done.
+
+Prepare Cilium to use AWS-CNI chaining
+======================================
+
+Download the Cilium deployment yaml:
+
+.. tabs::
+  .. group-tab:: K8s 1.14
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.14/cilium.yaml
+
+  .. group-tab:: K8s 1.13
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.13/cilium.yaml
+
+  .. group-tab:: K8s 1.12
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.12/cilium.yaml
+
+  .. group-tab:: K8s 1.11
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.11/cilium.yaml
+
+  .. group-tab:: K8s 1.10
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.10/cilium.yaml
+
+Edit ``cilium.yaml and add the following configuration to the ConfigMap:
+
+.. code:: bash
+
+      cni-chaining-mode: aws-cni
+      masquerade: "false"
+      tunnel: disabled
```

Comment:
Hmm I didn't check the code, for flannel I enforce the tunnel to be disabled and print a warning in the logs in case the user forgot to do that.

Reply:
What is your request? I do not want to make the agent aware of individual chaining modes. The agent is kept generic to support any plugin combination.

Reply:
Let me ask you differently, what are the consequences if the user does not set `disabled` in tunnel?

Reply:
It will perform unnecessary tunneling. It will still work though.